API_register/models.py

```diff
@@ -29,3 +36,8 @@ class Announcement(models.Model):
     author = models.ForeignKey(Teacher, on_delete=models.CASCADE)
     hidden = models.BooleanField(default=False)
     student_class = models.ForeignKey(Classes, on_delete=models.CASCADE, default=None, null=True, blank=True)
+
+    objects = AnnouncementManager()
```

Comment:
you also might need a reference to an original manager, for example in the admin panel.

```python
objects = AnnouncementManager()
original_objects = models.Manager()
```

Please update admin views.